sway-core/src/semantic_analysis/ast_node/expression/typed_expression/method_application.rs

```diff
@@ -337,13 +343,141 @@ pub(crate) fn type_check_method_application(
         is_method_call_syntax_used,
     )?;
 
+    let old_arguments = arguments;
     let arguments = method
         .parameters
         .iter()
         .map(|m| m.name.clone())
         .zip(args_buf.iter().cloned())
         .collect::<Vec<_>>();
 
+    if method.is_contract_call {
+        fn call_contract_call(
+            ctx: &mut TypeCheckContext,
+            return_type: TypeId,
+            method_name_expr: Expression,
+            caller: Expression,
+            arguments: Vec<Expression>,
+            typed_argumens: Vec<TypeId>,
+            coins_expr: Expression,
+            asset_id_expr: Expression,
+            gas_expr: Expression,
+        ) -> Expression {
+            let tuple_args_type_id = ctx.engines.te().insert(
+                ctx.engines,
+                TypeInfo::Tuple(
+                    typed_argumens
+                        .iter()
+                        .map(|&type_id| TypeArgument {
+                            type_id,
+                            initial_type_id: type_id,
+                            span: Span::dummy(),
+                            call_path_tree: None,
+                        })
+                        .collect(),
+                ),
+                None,
+            );
+            Expression {
+                kind: ExpressionKind::FunctionApplication(Box::new(
+                    FunctionApplicationExpression {
+                        call_path_binding: TypeBinding {
+                            inner: CallPath {
+                                prefixes: vec![],
```

Comment:
I assume we need to put `["core", "codec"]` here to get a unique and full call path. Otherwise, if there is a local function called "contract_call" that one will be resolved.

Reply:
Will also answer this one in the comments.